### PR TITLE
feat: Use `BaseNodeData` to add `tags` property to all component types

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Replaces the `MoreInformation` interface with `BaseNodeData`
   - Most of this PR is then just type changes that follow along
   - Removes duplicate properties (where `MoreInformation` was previously not being used)

## Why?
 - Part of the work to add tags to nodes
 - This new interface serves as a base for all Node types - meaning we always get both `NodeTags` and `MoreInformation`
 - Hopefully this pattern can be followed up fairly shortly to also pull in and standardise other common fields (`title`, `description`, maybe `fn`)

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/shared/index.ts
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/shared/index.ts:1]: This is the important file in this PR! 👇

File: editor.planx.uk/src/@planx/components/FileUploadAndLabel/schema.ts [hunk 1]
```diff
@@ -31,6 +31,10 @@ const moreInformationSchema: SchemaOf<MoreInformation> = object({
   definitionImg: string(),
 });
 
+const baseNodeDataSchema: SchemaOf<BaseNodeData> = object({
+  tags: array(mixed().oneOf(["placeholder"])),
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/FileUploadAndLabel/schema.ts:35]: nit: surprised to see `["placeholder"]` hardcoded specifically in this component file - can we define & export/import list of available tags someplace once?

**Review by jessicamcinchak — APPROVED** [review]
One small comment, but overall looks good to me :+1:

File: editor.planx.uk/src/@planx/components/FileUploadAndLabel/schema.ts (hunk 1 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/@planx/components/FileUploadAndLabel/schema.ts]: Good catch thanks - now added 👍

**Pull request merged**